plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1357 @@
+            requests.append(location_request)
+
+        if config.get('user'):
+            user_path = "data/ietf-snmp:snmp/usm/local/user"
+            payload = self.build_create_user_payload(config)
+            users_request = {'path': user_path, "method": method, 'data': payload}
+            requests.append(users_request)
+
+            vacm_path = "data/ietf-snmp:snmp/vacm"
+            vacm_payload = self.build_create_vacm_payload(config)
+            users_request = {'path': vacm_path, "method": method, 'data': vacm_payload}
+            requests.append(users_request)
+
+        if config.get('view'):
+            views_path = "data/ietf-snmp:snmp/vacm"
+            payload = self.build_create_view_payload(config)
+            views_request = {'path': views_path, 'method': method, 'data': payload}
+            requests.append(views_request)
+
+        return requests
+
+    def build_create_agentaddress_payload(self, config, have, overridden_or_replaced):
+        """ Build the payload for SNMP agentaddress
+        :rtype: A dictionary
+        :returns: The payload for SNMP agentaddress
+        """
+        agentaddress = config.get('agentaddress')
+        have_agentaddress = have.get('agentaddress')
+        have_agentaddress_names = []
+        if have_agentaddress and not overridden_or_replaced:
+            have_agentaddress_names = [agent_entry.get('name') for agent_entry in have_agentaddress]
+        agentaddress_list = []
+        agentaddressdict = {}
+        payload_url = {}
+        for conf in agentaddress:
+            name = ""
+            agentaddress_dict = {}
+            if 'name' in conf:
+                name = conf.get('name')
+            else:
+                name = self.get_agententry(have_agentaddress_names)
+
+            agentaddress_dict['name'] = name
+            agentaddress_dict['udp'] = {'ietf-snmp-ext:interface': conf.get('interface'), 'ip': conf.get('ip'), 'port': conf.get('port')}
+            agentaddress_list.append(agentaddress_dict)
+
+        agentaddressdict['listen'] = agentaddress_list
+        payload_url['ietf-snmp:engine'] = agentaddressdict
+        return payload_url
+
+    def build_create_community_payload(self, config, have):
+        """ Build the payload for SNMP community
+        :rtype: A dictionary
+        :returns: The list of community requests
+        """
+        community = config.get('community')
+        community_list = []
+        payload_url = {}
+        community_requests = []
+        community_path = "data/ietf-snmp:snmp/community"
+
+        for conf in community:
+            community_dict = {}
+            community_dict['index'] = conf.get('name')
+
+            if have and len(have) > 0 and have.get('group'):
+                for group in have.get(['group']):
+                    matched_group = next((each_group for each_group in group.get('member') if each_group['security-name'] == conf.get('name')), None)
```

Comment:
It isn't necessary to delete this community member from the "target" group if it is already a member of that group.

```suggestion
                    matched_group = next((each_group for each_group in group.get('member') if each_group['security-name'] == conf.get('name') and group.get('name') != conf.get('group')), None)
```